allennlp/tests/semparse/contexts/table_question_context_test.py

```diff
@@ -0,0 +1,103 @@
+    def setUp(self):
+        super().setUp()
+        self.tokenizer = WordTokenizer(SpacyWordSplitter(pos_tags=True))
+
+    def test_number_extraction(self):
+        question = """how many players on the 191617 illinois fighting illini men's basketball team
+                      had more than 100 points scored?"""
+        question_tokens = self.tokenizer.tokenize(question)
+        test_file = f'{self.FIXTURES_ROOT}/data/corenlp_processed_tables/TEST-7.table'
+        table_question_context = TableQuestionContext.read_from_file(test_file, question_tokens)
+        _, number_entities = table_question_context.get_entities_from_question()
+        assert number_entities == [("191617", "191617"), ("100", "100")]
+
+    def test_date_extraction(self):
+        question = "how many laps did matt kenset complete on february 26, 2006."
+        question_tokens = self.tokenizer.tokenize(question)
+        test_file = f'{self.FIXTURES_ROOT}/data/corenlp_processed_tables/TEST-8.table'
+        table_question_context = TableQuestionContext.read_from_file(test_file, question_tokens)
+        _, number_entities = table_question_context.get_entities_from_question()
+        assert number_entities == [("2", "february"), ("26", "26"), ("2006", "2006")]
+
+    def test_date_extraction_2(self):
+        question = """how many different players scored for the san jose earthquakes during their
+                      1979 home opener against the timbers?"""
+        question_tokens = self.tokenizer.tokenize(question)
+        test_file = f'{self.FIXTURES_ROOT}/data/corenlp_processed_tables/TEST-6.table'
+        table_question_context = TableQuestionContext.read_from_file(test_file, question_tokens)
+        _, number_entities = table_question_context.get_entities_from_question()
+        assert number_entities[0] == ("1979", "1979")
+
+    def test_multiword_entity_extraction(self):
+        question = "was the positioning better the year of the france venue or the year of the south korea venue?"
+        question_tokens = self.tokenizer.tokenize(question)
+        test_file = f'{self.FIXTURES_ROOT}/data/corenlp_processed_tables/TEST-3.table'
+        table_question_context = TableQuestionContext.read_from_file(test_file, question_tokens)
+        entities, _ = table_question_context.get_entities_from_question()
+        assert entities == ["france", "south_korea"]
+
+    def test_rank_number_extraction(self):
+        question = "what was the first tamil-language film in 1943?"
+        question_tokens = self.tokenizer.tokenize(question)
+        test_file = f'{self.FIXTURES_ROOT}/data/corenlp_processed_tables/TEST-1.table'
+        table_question_context = TableQuestionContext.read_from_file(test_file, question_tokens)
+        _, numbers = table_question_context.get_entities_from_question()
+        assert numbers[0] == ("1", "first")
+
+    def test_null_extraction(self):
+        question = "on what date did the eagles score the least points?"
```

Comment:
"eagles" doesn't appear because it's the title of the wikipedia page, or something?  A note about that would be helpful here, as this test is confusing without it.

Reply:
Done.